src/app/asset-grid/asset-grid.component.pug

```diff
@@ -123,9 +123,9 @@
                 [largeThmbView]="largeThmbView"
               )
             .mobile--hide
-              input.check(tabIndex="3", type="checkbox", [checked]="isSelectedAsset(asset) > -1", [ngStyle]="{ 'opacity': isSelectedAsset(asset) > -1 ? '1' : '0' }", (change)="editMode = true; selectAsset(asset);")
+              input.check([attr.aria-label]="'Select asset \"' + asset.name + '\"'", tabIndex="3", type="checkbox", [checked]="isSelectedAsset(asset) > -1", [ngStyle]="{ 'opacity': isSelectedAsset(asset) > -1 ? '1' : '0' }", (change)="editMode = true; selectAsset(asset);")
             .mobile--show
-              input.check-mobile(type="checkbox", [checked]="isSelectedAsset(asset) > -1", (change)="editMode = true; selectAsset(asset);")
```

Comment:
This is real ugly. Please lmk if there is a better way to do the label construction.

Reply:
Would it be overkill to extract the string building into a function/variable on `asset-grid.component`? Might be able to use ES6 string templates there too?

Reply:
I think Evan is right, I would put this into a function. Easier to maintain if it has to be changed in the future.

Reply:
Done!